### PR TITLE
Feat: Local Event Repository Logic + Tests

### PR DESCRIPTION
**Pull request by pukoa420**
Added:
- Robust Error Handling with **Result<T>** (thanks for the suggestion @AlecGhost)
- Enforced Data Integrity in _createEvent_ by preventing creating duplicate id events
- Centralized test data of events
- Added Unit tests for all local event repo

Refactor:
 - _createEvent(), updateEvent(), deleteEvent()_ now return Result<T> to indicate Sueccess/Failure
 - changed local+firebase repo to fit new interface
 - changed docs according to changes

Testing:
- tests are checking all basic functions as well as certain edge cases
- almost full code coverage
image

To consider:
- We should try to generally centralize test data as I did here because of heave code duplication as experienced last meeting

### CODE REVIEW

File: app/src/main/java/ch/epfllife/model/event/EventRepository.kt [hunk 1]
```diff
@@ -19,27 +19,34 @@ interface EventRepository {
...
+  suspend fun getEvent(eventId: String): Event?
 
   /**
    * Adds a new Event item to the repository.
    *
    * @param event The Event item to add.
+   * @return A result indicating the success or failure of the operation.
    */
-  suspend fun createEvent(event: Event)
+  suspend fun createEvent(event: Event): Result<Unit>
```
Comment (GustavOnsberg) [inline app/src/main/java/ch/epfllife/model/event/EventRepository.kt]: Is there a reason not to return a bool or enum to indicate success? I would thing that would be less ambitious

File: app/src/main/java/ch/epfllife/model/event/EventRepositoryFirestore.kt [hunk 2]
```diff
@@ -27,16 +27,16 @@ class EventRepositoryFirestore(private val db: FirebaseFirestore) : EventReposit
...
   }
 
-  override suspend fun updateEvent(eventId: String, newEvent: Event) {
+  override suspend fun updateEvent(eventId: String, newEvent: Event): Result<Unit> {
     // add a check if the eventId is same as newEvent.Id
     TODO("Not yet implemented")
   }
 
-  override suspend fun deleteEvent(eventId: String) {
+  override suspend fun deleteEvent(eventId: String): Result<Unit> {
```
Comment (GustavOnsberg) [inline app/src/main/java/ch/epfllife/model/event/EventRepositoryFirestore.kt]: same

File: app/src/main/java/ch/epfllife/model/event/EventRepositoryLocal.kt [hunk 3]
```diff
@@ -1,28 +1,59 @@
...
 
-  override suspend fun getEvent(eventId: String): Event {
-    TODO("Not yet implemented")
+  override suspend fun getEvent(eventId: String): Event? {
+    return events.find { it.id == eventId }
   }
 
-  override suspend fun createEvent(event: Event) {
-    TODO("Not yet implemented")
+  override suspend fun createEvent(event: Event): Result<Unit> {
```
Comment (GustavOnsberg) [inline app/src/main/java/ch/epfllife/model/event/EventRepositoryLocal.kt]: Should we even have createEvent on the local repository? I would think it would be simpler and cleaner to only allow for event creation directly to firebase. Then that should also take care of creating the ids

File: app/src/main/java/ch/epfllife/model/event/EventRepository.kt (hunk 1 above)
Comment (pukoa420) [reply app/src/main/java/ch/epfllife/model/event/EventRepository.kt]: The Result<T> acts as a enum where you can can call onSuccess and onFailure(). This was discussed with Alex

File: app/src/main/java/ch/epfllife/model/event/EventRepositoryFirestore.kt [hunk 4]
```diff
@@ -27,16 +27,16 @@ class EventRepositoryFirestore(private val db: FirebaseFirestore) : EventReposit
     TODO("Not yet implemented")
   }
 
-  override suspend fun createEvent(event: Event) {
+  override suspend fun createEvent(event: Event): Result<Unit> {
     TODO("Not yet implemented")
   }
 
-  override suspend fun updateEvent(eventId: String, newEvent: Event) {
+  override suspend fun updateEvent(eventId: String, newEvent: Event): Result<Unit> {
```
[the remark this answers is not in the document]
Comment (pukoa420) [reply app/src/main/java/ch/epfllife/model/event/EventRepositoryFirestore.kt]: see first comment

File: app/src/main/java/ch/epfllife/model/event/EventRepositoryFirestore.kt (hunk 2 above)
Comment (pukoa420) [reply app/src/main/java/ch/epfllife/model/event/EventRepositoryFirestore.kt]: see first comment

File: app/src/main/java/ch/epfllife/model/event/EventRepositoryLocal.kt (hunk 3 above)
Comment (pukoa420) [reply app/src/main/java/ch/epfllife/model/event/EventRepositoryLocal.kt]: The local repo is **only** for testing. So we need this. The firebase repo is independant.
